enhancements/installer/aws-cross-account-dns-zone.md

```diff
@@ -0,0 +1,260 @@
+  - https://issues.redhat.com/browse/OCPBU-558
+see-also:
+replaces:
+superseded-by:
+---
+
+# AWS Shared VPC with Cross-account DNS Zones
+
+## Summary
+
+AWS [VPC sharing](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-sharing.html) allows a VPC
+to be shared between multiple accounts. Accordingly, it is possible to configure a DNS private hosted
+zone in one account (Account A) while cluster resources are installed in a different
+account (Account B). This enhancement enables OpenShift installation with this cross-account shared-VPC
+architecture by providing a solution to handle authentication to the private-hosted-zone account. 
+
+## Motivation
+
+Organizations utilize AWS shared VPCs to enable some degree of resource sharing with participants while
+maintaining administrative privileges over others. Enabling users to install OpenShift with a private
+hosted zone from a separate account allows DNS resolution for the cluster on the shared network while
+maintaining the least-privilege security principle. An alternative solution would be for the OpenShift
+Installer to create a private hosted zone and attach the zone to the shared VPC. Instead, organizations
+can limit the required permissions and maintain greater administrative control by pre-creating a hosted
+zone and a role with limited permissions in that secondary account.
+
+### User Stories
+
+* As an OpenShift cluster admin, I want to install OpenShift on a shared VPC and utilize a private hosted
+zone that belongs to an account different than the rest of the cluster resources.
+
+### Goals
+
+Utilizing the already existing `platform.aws.hostedZone` field in the install config, users
+should be able to install when passing an ID for a hosted zone that belongs to a different account
+than the primary credentials for installation.
+
+### Non-Goals
+
+Modifying public DNS zones from separate accounts is out of scope.
+
+## Proposal
+
+Cluster admins should create a role in the private hosted zone account (Account A),
+and pass the ARN for that role in the install config. The presence of this roleARN
+indicates to the installer/cluster that the private hosted zone belongs to a different
+account, and that the specified role should be assumed when performing operations on
+the hosted zone.
+
+![alt Architecture diagram](aws_shared_vpc.png)
+
+
+### Workflow Description
+
+**cluster creator** is a human user who has been granted **Account B IAM User**
+
+**shared vpc admin** is a human user with super-user-like privileges across multiple accounts.
+
+1. **shared vpc admin** shares a VPC with Account B.
+2. **shared vpc admin** creates a private hosted zone in Account A and attaches it to the shared VPC.
+3. **shared vpc admin** creates a role in Account A granting **Account B IAM User** route53 and tagging
+permissions in Account A.
+4. **cluster creator** specifies the role from step 3 and subnets from step 1 in the install config.
+5. **cluster creator** installs the cluster.
```

Comment:
Suggested updates expanding the IAM details with examples:

> Sorry I can't use the `suggestion` feature as it was breaking the code blocks

---

1. **shared vpc admin** shares [VPC subnets](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-sharing.html) with Account B.
2. **shared vpc admin** creates a private hosted zone in Account A and associates it to the shared VPC.
3. **shared vpc admin** creates an IAM policy in Account A granting route53 and tagging
permissions to the private hosted zone. Example:
~~~json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "route53:GetHostedZone",
                "route53:ChangeResourceRecordSets",
                "route53:ChangeTagsForResource"
            ],
            "Resource": [
                "arn:aws:route53:::hostedzone/Z00147933I3NWOQ6M4699"
            ]
        }
    ]
}
~~~
4. **shared vpc admin** creates an IAM role in Account A with Trust Policy granting **Account B IAM User** to assume it, and attach the custom IAM policy created on step 3. Example of Trust Policy:
~~~json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::[ID for Account B]:user/<user name for cluster creator>"
            },
            "Action": "sts:AssumeRole",
            "Condition": {}
        }
    ]
} 
~~~
6. **cluster creator** specifies the IAM role from step 3 and subnets from step 1 in the install config.
7. **cluster creator** installs the cluster.

Reply:
:+1: I think the additional details are worth capturing in the workflow description.

Reply:
Great suggestion. Thanks Marco.